src/dmd/parse.d

```diff
@@ -4155,15 +4175,15 @@ final class Parser(AST) : Lexer
                             AST.Type index = parseType(); // [ type ]
                             check(TOK.rightBracket);
                             ta = new AST.TypeAArray(t, index);
-                            *palt |= 2;
+                            palt |= 2;
                         }
                         else
                         {
                             //printf("It's a static array\n");
                             AST.Expression e = parseAssignExp(); // [ expression ]
                             ta = new AST.TypeSArray(t, e);
                             check(TOK.rightBracket);
-                            *palt |= 2;
+                            palt |= 2;
```

Comment:
`palt |= 2;` is executed unconditionally in every branch of `case TOK.leftBracket`